comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,42 +10,130 @@ public enum Mark { None, O, X };
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            int tilesUsed = 0;
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            // Checks every square on the board to see if there are any empty ones.
+            // If there aren't any, tilesUsed is used as a check.
+            for (int i = 0; i < 3; i++)
+            {
+                for (int j = 0; j < 3; j++)
+                {
+                    Mark currentMark = oxoArray[i, j];
+                    if (currentMark == Mark.None)
+                    {
+                        return false;
+                    }
+                    else
+                    {
+                        tilesUsed++;
+                    }
+                }
+            }
+            // Check to see if there are any tiles left according to the variable tilesUsed.
+            if(tilesUsed < 9)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
 		}
 
-		// Display the current board state in the terminal. You should only need to edit this if you are attempting the stretch goal.
-		public void PrintBoard()
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            if (oxoArray[0, 0] == Mark.X && oxoArray[1, 0] == Mark.X && oxoArray[2, 0] == Mark.X)
+            {
+                return Mark.X;
+            }
+            else if (oxoArray[0, 0] == Mark.X && oxoArray[1, 1] == Mark.X && oxoArray[2, 1] == Mark.X)
+            {
+                return Mark.X;
+            }
+            else if (oxoArray[0, 2] == Mark.X && oxoArray[1, 2] == Mark.X && oxoArray[2, 2] == Mark.X)
+            {
+                return Mark.X;
+            }
+
+
+            else if (oxoArray[0, 0] == Mark.O && oxoArray[1, 0] == Mark.O && oxoArray[2, 0] == Mark.O)
+            {
+                return Mark.O;
+            }
+            else if (oxoArray[0, 0] == Mark.O && oxoArray[1, 1] == Mark.O && oxoArray[2, 1] == Mark.O)
+            {
+                return Mark.O;
+            }
+            else if (oxoArray[0, 2] == Mark.O && oxoArray[1, 2] == Mark.O && oxoArray[2, 2] == Mark.O)
+            {
+                return Mark.O;
+            }
+
+            else if (oxoArray[0, 0] == Mark.X && oxoArray[1, 1] == Mark.X && oxoArray[2, 2] == Mark.X)
+            {
+                return Mark.X;
+            }
+            else if (oxoArray[0, 2] == Mark.X && oxoArray[1, 1] == Mark.X && oxoArray[2, 0] == Mark.X)
+            {
+                return Mark.X;
+            }
+
```

Comment:
It looks like you forgot to check vertical lines -- you are only checking horizontal and diagonal! Hence why you have failing unit tests